### PR TITLE
move imports from react-query/hydration to react-query

### PR DESCRIPTION
**Pull request by lukahartwig**
As requested [here ](https://github.com/trpc/trpc/pull/1836) I opened a PR from within the repo

### CODE REVIEW

File: packages/next/src/withTRPC.tsx [hunk 1]
```diff
@@ -18,8 +18,13 @@ import {
   NextPageContext,
 } from 'next/dist/shared/lib/utils';
 import React, { createElement, useState } from 'react';
-import { QueryClient, QueryClientProvider } from 'react-query';
-import { DehydratedState, Hydrate, dehydrate } from 'react-query/hydration';
+import {
+  DehydratedState,
+  Hydrate,
```
Comment (KATT) [inline packages/next/src/withTRPC.tsx:23]: Do you know at which minor this was changed in react-query 3? We would need to update `peerDependencies` to that version as well in

**Review by KATT — COMMENTED** [review]
Thanks for this! Looks great, just left one question

**Comment by KATT** [comment]
@all-contributors add @lukahartwig for code

File: packages/next/src/withTRPC.tsx (hunk 1 above)
Comment (lukahartwig) [reply packages/next/src/withTRPC.tsx:23]: The change was made in https://github.com/tannerlinsley/react-query/releases/tag/v3.22.0. I double checked the `peerDependencies`, all of them are above that.

Comment (KATT) [reply packages/next/src/withTRPC.tsx:23]: Perfect

**Review by KATT — APPROVED** [review]

**Pull request merged**